deducing-this.md

```diff
@@ -636,14 +714,14 @@ template <typename T>
 class optional {
   // ...
   template <typename Self>
-  constexpr auto&& operator*(Self&& this self) {
-    return forward<Self>(self).m_value;
+  constexpr auto&& operator*(Self&& this) {
+    return forward<like_t<Self, optional>>(*this).m_value;
```

Comment:
Why do we need to use the `like_t` version here?

Reply:
If anyone inherits from optional you want to be sure you're accessing the correct member.